snap/snapcraft.yaml

```diff
@@ -12,256 +13,318 @@ description: |
+    daemon: oneshot
     environment:
       PKI_SETUP_VAULT_FILE: $SNAP_DATA/config/security-secret-store/pkisetup-vault.json
       PKI_SETUP_KONG_FILE: $SNAP_DATA/config/security-secret-store/pkisetup-kong.json
-    daemon: oneshot
+    passthrough:
+      start-timeout: 15m
   vault:
-    command: bin/vault server --config="$VAULT_CONFIG"
+    adapter: none
+    after:
+      - consul
+      - pkisetup
+      - core-config-seed
+    command: bin/vault server --config $VAULT_CONFIG
+    daemon: simple
     environment:
-      VAULT_CONFIG: "$SNAP_DATA/config/security-secret-store/vault-config.json"
+      VAULT_CONFIG: "$SNAP_DATA/config/security-secret-store/vault-config.hcl"
       VAULT_ADDR: "https://localhost:8200"
       VAULT_UI: "true"
-    daemon: simple
     plugs:
       - network
       - network-bind
-    passthrough:
-      after:
-        - consul
-        - pkisetup
-        - core-config-seed
   vault-worker:
+    adapter: full
+    after: [vault]
     command: bin/vault-worker-wrapper.sh
     daemon: oneshot
+    passthrough:
+      start-timeout: 15m
     plugs:
       - network
       - network-bind
-    passthrough:
-      after:
-        - vault
   edgexproxy:
+    adapter: none
+    after:
+      - vault-worker
+      - kong-daemon
     command: bin/edgexproxy-wrapper.sh
     daemon: oneshot
+    passthrough:
+      start-timeout: 15m
     plugs:
       - network
       - network-bind
-    passthrough:
-      after:
-        - vault-worker
-        - kong-daemon
   mongo-worker:
+    adapter: full
+    after: [mongod]
     command: bin/mongo-worker-wrapper.sh
-    plugs: [network]
     daemon: oneshot
     passthrough:
-      after:
-        - mongod
-  support-logging:
-    command: bin/fork-wrapper.sh $SNAP/bin/go-services-wrapper.sh support-logging --registry
-    daemon: forking
-    plugs: [network, network-bind]
-    passthrough:
-      after:
-        - core-config-seed
-        - mongo-worker
-        - edgexproxy
-  support-notifications:
-    command: bin/fork-wrapper.sh $SNAP/bin/go-services-wrapper.sh support-notifications --registry
-    daemon: forking
-    plugs: [network, network-bind]
-    passthrough:
-      after:
-        - core-config-seed
-        - mongo-worker
-        - edgexproxy
+      start-timeout: 15m
+    plugs: [network]
   core-data:
-    command: bin/fork-wrapper.sh $SNAP/bin/go-services-wrapper.sh core-data --registry
-    daemon: forking
+    adapter: full
+    after:
+      - core-config-seed
+      - mongo-worker
+      - edgexproxy
+    command: bin/core-data -confdir $SNAP_DATA/config/core-data/res --registry
+    daemon: simple
     plugs: [network, network-bind]
-    passthrough:
-      after:
-        - core-config-seed
-        - mongo-worker
-        - edgexproxy
+    # for now, specify stop-timeout as really short until services learn how
+    # to handle SIGTERM properly with as per 
+    # https://github.com/edgexfoundry/edgex-go/issues/1306
+    stop-timeout: 1s
```

Comment:
This seems short for a kill timeout for a process that deals with a db. Any chance this could lead to data corruption if core-data were in the midst of flushing writes to the db?

Reply:
It could lead to data corruption. What do you think is more reasonable? Without this, systemd will use the snapd default of 30 seconds, and due to the referenced bug the service doesn't respond to SIGTERM at all in that time meaning that we always end up waiting at least 30 seconds to remove the snap or stop any of the services.

Reply:
10s? Honestly until the services are fixed to respond properly to SIGTERM, that's still a blind guess, and corruption could still possibly occur, but it seemed like 1s was too short...

Reply:
Okay, I changed it to 10s